browser/src/core/Socket.js

```diff
@@ -833,12 +833,12 @@ app.definitions.Socket = L.Class.extend({
 							                                      this.value = 'overwrite';
 							                                      this.close();
 						                                      }}),
-						$.extend({}, vex.dialog.buttons.YES, { text: _('Save to new file'),
+						(that._map['wopi'].UserCanNotWriteRelative ? undefined : $.extend({}, vex.dialog.buttons.YES, { text: _('Save to new file'),
 						                                      click: function() {
 							                                      this.value = 'saveas';
 							                                      this.close();
-						                                      }})
-					],
+						                                      }}))
+					].filter(Boolean),
```

Comment:
I think this code is a bit complicated. To make it more readable maybe we can construct that array before `vex.dialog.open` call.
eg.
```
var buttons = [ <first entry>, <second entry> ];
if (!that._map['wopi'].UserCanNotWriteRelative)
    buttons.push( <optional entry> )
```